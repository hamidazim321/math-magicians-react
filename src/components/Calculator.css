.calculator {
  display: grid;
  grid-template-columns: auto auto auto auto;
  width: 35%;
}

.calculator-screen {
  grid-column: 1/span 4;
  height: 60px;
  background-color: rgb(133, 117, 117);
  border: solid 1px grey;
  font-weight: bold;
  font-size: 32px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.btn {
  height: 60px;
  border: solid 1px grey;
  background-color: silver;
  font-weight: 600;
  font-size: 16px;
}

.btn-zero {
  grid-column: 1/ span 3;
}

.btn-red {
  background-color: rgb(230, 113, 35);
}
