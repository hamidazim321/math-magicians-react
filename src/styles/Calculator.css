.calculator {
  display: grid;
  grid-template-columns: auto auto auto auto;
  width: 45%;
  margin-left: auto;
  margin-right: 50px;
}

.calculator-screen {
  grid-column: 1/span 4;
  height: 65px;
  background-color: rgb(133, 117, 117);
  border: solid 1px grey;
  font-weight: bold;
  font-size: 32px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.btn {
  height: 65px;
  border: solid 1px grey;
  background-color: silver;
  font-weight: 600;
  font-size: 16px;
}

.btn-zero {
  grid-column: 1/ span 2;
}

.btn-red {
  background-color: rgb(230, 113, 35);
}
