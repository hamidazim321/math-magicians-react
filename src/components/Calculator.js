import React, { useState } from 'react';
import './Calculator.css';
import CalcBtn from './Btn';
import calculate from '../logic/Calculate';

function Calculator() {
  const [calcObj, updateCalcObj] = useState(
    {
      operation: null,
      total: null,
      next: null,
    },
  );

  const handleClick = (btnName) => {
    updateCalcObj(calculate(calcObj, btnName));
  };

  return (
    <div className="calculator">
      <div className="calculator-screen">
        {calcObj.total}
      </div>
      <CalcBtn text="AC" handleClick={handleClick} />
      <CalcBtn text="+/-" handleClick={handleClick} />
      <CalcBtn text="%" handleClick={handleClick} />
      <CalcBtn text="รท" handleClick={handleClick} />
      <CalcBtn text="7" handleClick={handleClick} />
      <CalcBtn text="8" handleClick={handleClick} />
      <CalcBtn text="9" handleClick={handleClick} />
      <CalcBtn text="x" handleClick={handleClick} />
      <CalcBtn text="4" handleClick={handleClick} />
      <CalcBtn text="5" handleClick={handleClick} />
      <CalcBtn text="6" handleClick={handleClick} />
      <CalcBtn text="-" handleClick={handleClick} />
      <CalcBtn text="1" handleClick={handleClick} />
      <CalcBtn text="2" handleClick={handleClick} />
      <CalcBtn text="3" handleClick={handleClick} />
      <CalcBtn text="+" handleClick={handleClick} />
      <CalcBtn text="0" handleClick={handleClick} />
      <CalcBtn text="=" handleClick={handleClick} />
    </div>
  );
}

export default Calculator;
