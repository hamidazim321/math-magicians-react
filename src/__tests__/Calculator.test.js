import {
  render, fireEvent, screen, cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom';

import Calculator from '../components/Calculator';

afterEach(() => {
  cleanup();
});

describe('Integration tests for calculator component', () => {
  test('should render', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('2 + 2 should display 4', async () => {
    render(<Calculator />);

    const btnTwo = await screen.findByText(/2/);
    const btnPlus = await screen.findByText('+');
    const btnEqual = await screen.findByText(/=/);

    await fireEvent.click(btnTwo);
    await fireEvent.click(btnPlus);
    await fireEvent.click(btnTwo);
    await fireEvent.click(btnEqual);

    const calcScreen = document.querySelector('.calculator-screen p');

    expect(await calcScreen.textContent).toBe('4');
  });
});
