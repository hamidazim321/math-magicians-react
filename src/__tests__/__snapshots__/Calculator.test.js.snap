// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration tests for calculator component should render 1`] = `
<DocumentFragment>
  <h1>
    Lets Do Some Math!
  </h1>
  <div
    class="calculator"
  >
    <div
      class="calculator-screen"
    >
      <p />
    </div>
    <button
      class="btn "
      type="button"
    >
      AC
    </button>
    <button
      class="btn "
      type="button"
    >
      +/-
    </button>
    <button
      class="btn "
      type="button"
    >
      %
    </button>
    <button
      class="btn btn-red"
      type="button"
    >
      รท
    </button>
    <button
      class="btn "
      type="button"
    >
      7
    </button>
    <button
      class="btn "
      type="button"
    >
      8
    </button>
    <button
      class="btn "
      type="button"
    >
      9
    </button>
    <button
      class="btn btn-red"
      type="button"
    >
      x
    </button>
    <button
      class="btn "
      type="button"
    >
      4
    </button>
    <button
      class="btn "
      type="button"
    >
      5
    </button>
    <button
      class="btn "
      type="button"
    >
      6
    </button>
    <button
      class="btn btn-red"
      type="button"
    >
      -
    </button>
    <button
      class="btn "
      type="button"
    >
      1
    </button>
    <button
      class="btn "
      type="button"
    >
      2
    </button>
    <button
      class="btn "
      type="button"
    >
      3
    </button>
    <button
      class="btn btn-red"
      type="button"
    >
      +
    </button>
    <button
      class="btn btn-zero "
      type="button"
    >
      0
    </button>
    <button
      class="btn btn-red"
      type="button"
    >
      =
    </button>
  </div>
</DocumentFragment>
`;
